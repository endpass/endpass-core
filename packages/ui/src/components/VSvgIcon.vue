<template>
  <svg
    v-if="iconUrl"
    :class="className"
    :width="width"
    :height="height"
    :fill="fill"
    xmlns="http://www.w3.org/2000/svg"
  >
    <use
      :xlink:href="iconUrl"
      xmlns:xlink="http://www.w3.org/1999/xlink"
    />
  </svg>
</template>

<script>
export default {
  name: 'VSvgIcon',
  props: {
    width: {
      type: String,
      default: '20px',
    },
    height: {
      type: String,
      default: '20px',
    },
    fill: {
      type: String,
      default: '',
    },
    iconPath: {
      validator(value) {
        return !!value.default;
      },
    },
  },
  data: () => ({
    className: 'v-svg-icon',
  }),
  computed: {
    iconUrl() {
      const url = ((this.iconPath || {}).default || {}).url || null;
      return url;
    },
  },
};
</script>
<style lang="postcss">
.v-svg-icon {
  margin: 0 auto;
}
</style>
