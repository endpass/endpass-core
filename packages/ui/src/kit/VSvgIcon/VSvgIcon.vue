<script>
import SvgAtom from '@/atom/svg-atom/svg-atom.vue';

export default {
  name: 'VSvgIcon',

  extends: SvgAtom,
};
</script>
