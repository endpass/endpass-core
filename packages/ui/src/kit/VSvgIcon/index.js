export { default } from './VSvgIcon.vue';
