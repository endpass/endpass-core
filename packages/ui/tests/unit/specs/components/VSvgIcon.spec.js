import { shallowMount } from '@vue/test-utils';

import VSvgIcon from '@/components/VSvgIcon';

describe('VSvgIcon', () => {
  let wrapper;
  const url = 'TestUrl';

  beforeAll(() => {
    wrapper = shallowMount(VSvgIcon, {
      propsData: {
        iconPath: {
          default: {
            url,
          },
        },
      },
    });
  });

  describe('render', () => {
    it('should be a Vue component', () => {
      expect(wrapper.name()).toBe('VSvgIcon');
      expect(wrapper.isVueInstance()).toBeTruthy();
    });

    it('should render component', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });
  });

  describe('props', () => {
    it('should use default props', () => {
      const svg = wrapper.find('svg');

      expect(svg.attributes('width')).toBe('20px');
      expect(svg.attributes('height')).toBe('20px');
      expect(svg.attributes('fill')).toBe('');
      expect(svg.find('use').attributes('href')).toBe(url);
    });

    it('should redefine props', () => {
      wrapper.setProps({
        width: '125px',
        height: '256px',
        fill: 'blackWindow',
        iconPath: { default: { url: 'newPath' } },
      });

      const svg = wrapper.find('svg');

      expect(svg.attributes('width')).toBe('125px');
      expect(svg.attributes('height')).toBe('256px');
      expect(svg.attributes('fill')).toBe('blackWindow');
      expect(svg.find('use').attributes('href')).toBe('newPath');
    });
  });
});
