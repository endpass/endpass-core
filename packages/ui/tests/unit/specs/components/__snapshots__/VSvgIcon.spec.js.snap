// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VSvgIcon render should render component 1`] = `
<svg width="20px" height="20px" fill="" xmlns="http://www.w3.org/2000/svg" class="v-svg-icon">
  <use xlink:href="TestUrl" xmlns:xlink="http://www.w3.org/1999/xlink"></use>
</svg>
`;
